### DISCUSSION TITLE
Asphere surface example

### DISCUSSION THREAD

**Discussion by mathieupaurisse** [mjhoptics/ray-optics#41] [Q&A]
Hello,
First of all a big thank you for the amazing work you've been doing on this project so far. 
I am trying to simulate a small optical system containing an even asphere lens, and I am struggling to make it work.
The definition of the aspherical profile is ok, I used the following :
`asph_prof = profiles.EvenPolynomial(c = 1.506820634376125345E-01,
                                    cc = -0.5500111879614999,
                                    coefs = [0, 2.6715670411349998e-06,
                                             -4.1960626180280001e-07, 0,
                                            0, 0, 0, 0])`

But I do not understand how to make it a surface that I can add into my sequential model. What syntax should I use for that purpose?

I tried using `insert(s, g)` where g is the glass gap and s is the previous profile, but it does not work as `asph_prof` is not a surface.

Can an example be provided for such a situation?

Thank you!

Mathieu

**Comment by mjhoptics** [comment]
Hello Mathieu,
There's an example in my notebook repo, [ray-optics-notebooks](https://github.com/mjhoptics/ray-optics-notebooks/blob/master/Cell%20Phone%20lens.ipynb). The example uses RadialPolynomials but the idea is the same.

If I have a lens prescription in surface-by-surface form, I use the add_surface method in SequentialModel:
`sm.add_surface([.1507, thick, glass/catalog or index/v_number])`

I then set the profile attribute of the newly created surface with my aspheric profile definition.
`sm.ifcs[sm.cur_surface].profile = asph_profile`

Hope this helps. Mike
